### DISCUSSION TITLE
Responsive animations

### DISCUSSION THREAD

**Discussion by jackmurraybrown** [unpoly/unpoly#632] [Q&A]
What would the best way to implement responsive animations be? So for example you wanted the animation for a modal on mobile to slide from the bottom (could be using predefined `move-from-bottom`) but on desktop you just want the modal to fade in (could be using predefined `fade-in).

**Comment by triskweline** [comment]
You could define a custom animation that measures the screen width and picks an animation:

```js
up.animation('responsive-appear', function(element, options) {
  let isMobile = window.innerWidth < 500
  let animation = isMobile ? 'move-from-bottom' : 'fade-in'
  return up.animate(element, animation, options)
})
```
